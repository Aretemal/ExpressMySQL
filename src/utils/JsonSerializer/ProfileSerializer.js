import Serializer from './Serializer.js';

class ProfileSerializer extends Serializer {
  getInfoAuthorizedUser(data, links, id) {
    return this.serialize(data, id, 'ObjectUser', links);
  }

  updateStatus(data, links, id) {
    return this.serialize(data, id, 'NewStatus', links);
  }
}
export default new ProfileSerializer();
