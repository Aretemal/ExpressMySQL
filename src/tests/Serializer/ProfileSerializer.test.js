import ProfileSerializer from '../../utils/JsonSerializer/ProfileSerializer.js';

describe('ProfileSerializer :', () => {
  describe('getInfoAuthorizedUser :', () => {
    test('should return info about user', () => {
      const links = 'http://localhost:5000/api/profile/user';
      const user = {
        id: 1,
        login: 'Artem',
        firstName: 'aaaa',
        lastName: 'bbbb',
        email: '111@11.1',
      };

      const data = ProfileSerializer.getInfoAuthorizedUser(user, links, 1);

      expect(data).toMatchSnapshot();
    });
  });

  describe('updateStatus :', () => {
    test('should return new status', () => {
      const links = 'http://localhost:5000/api/profile/status';
      const status = { status: 'Hello world' };

      const data = ProfileSerializer.updateStatus(status, links, 1);

      expect(data).toMatchSnapshot();
    });
  });
});
