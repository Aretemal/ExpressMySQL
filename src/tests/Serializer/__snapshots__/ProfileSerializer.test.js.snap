// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileSerializer : getInfoAuthorizedUser : should return info about user 1`] = `
{
  "data": {
    "attributes": {
      "email": "111@11.1",
      "firstName": "aaaa",
      "id": 1,
      "lastName": "bbbb",
      "login": "Artem",
    },
    "id": 1,
    "type": "ObjectUser",
  },
  "links": {
    "self": "http://localhost:5000/api/profile/user",
  },
}
`;

exports[`ProfileSerializer : updateStatus : should return new status 1`] = `
{
  "data": {
    "attributes": {
      "status": "Hello world",
    },
    "id": 1,
    "type": "NewStatus",
  },
  "links": {
    "self": "http://localhost:5000/api/profile/status",
  },
}
`;
